griptape/mixins/activity_mixin.py

```diff
@@ -3,16 +3,24 @@
 import inspect
 from typing import Callable, Optional
 
-import schema
 from attrs import Attribute, define, field
 from jinja2 import Template
-from schema import Literal, Schema
+from schema import Schema
 
 
 @define(slots=False)
 class ActivityMixin:
+    """Provides Tool Activity management functionality to Tools.
+
+    Attributes:
+        allowlist: List of Tool Activities to include in the Tool schema.
+        denylist: List of Tool Activities to remove from the Tool schema.
+        extra_schema_properties: Mapping of Activity name and extra properties to include in the activity's schema.
+    """
+
     allowlist: Optional[list[str]] = field(default=None, kw_only=True)
     denylist: Optional[list[str]] = field(default=None, kw_only=True)
+    extra_schema_properties: dict = field(default=None, kw_only=True)
```

Comment:
Take a look at the (previously missing) example, the API takes the _value_ of a `Schema` object which is [hard to type](https://github.com/keleshev/schema/blob/master/schema/__init__.py#L358). Should we take in a `Schema` instead? The reason I avoided this is because we're only using the value of the Schema -- throwing away things like `Schema.description`.

Either way,  I narrowed the scope a bit to `dict[str, dict]`